### ISSUE TITLE
Environment variables

### ISSUE THREAD

**Issue by KaKi87** [files-community/files-uwp#182]
Hello,

I would like to be able to use environment variables in the path bar, like :

- `%appdata%`
- `%localappdata%`
- `%temp%`
- `%userprofile%`
- `%programfiles%`
- `%programdata%`

Etc.

Thanks

**Comment by KaKi87** [comment]
@yaichenbaum You asked me to try using Files UWP regularly so I can report my issues and feature requests, but it's not useful since all my issues got *won't fix* or *low priority*.
I guess we're done here.
++

**Comment by yaichenbaum** [comment]
Hey I am sorry you feel that I am not taking your feedback seriously, that couldn't be further from the truth, even the ones marked as wont fix will be revisited after some time, the feedback you provided is extremely valuable, low priority means whats that it is not absolutely necessary for the average user but they are still things that are extremely important and we will try hard to get to before Launch 2020. Thank you for the time you spent trying out Files UWP.

**Comment by duke7553** [comment]
Hey there @KaKi87 ,

Files is in the process of nailing the fundamentals. While quite a bit of work has been done in this area, there are some things that need to be worked on first before there is any focus on refinement. I can't state enough how thankful I am for you taking the time to self-host this project and report back to us what needs to be fixed. Realize that priority labels don't mean that we're not going to fix the issue any time soon. They just serve as a reminder of how important an issue really is to our goal of "nailing the fundamentals first." If you take a peek at the issues that have been fixed, a good portion of them were actually marked "low-priority."

However, I understand your frustration and will be re-thinking the "low-priority" label as it is a probable misnomer on my part.

Thank you for your understanding.

**Comment by yaichenbaum** [comment]
Changes made in https://github.com/duke7553/files-uwp/pull/194 should address this issue.

**Comment by yaichenbaum** [comment]
@KaKi87 which of the variables can change, happy to fix up the pr to correctly support this request.

**Comment by KaKi87** [comment]
Windows environment variables can be manipulated as desired : add, remove, edit.
http://environmentvariables.org/Main_Page

**Comment by yaichenbaum** [comment]
So if I understand correctly it would need to read the registry values. Thanks for the link :)

**Comment by yaichenbaum** [comment]
Is `C:\Program Files` always the same or can it be changed as well?

**Comment by yaichenbaum** [comment]
We should take a look at https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariables?view=netframework-4.8 for full support.

**Comment by duke7553** [comment]
@yaichenbaum When you get a chance, I think we should determine how much work on this is left.

**Comment by yaichenbaum** [comment]
@duke7553 There is still quite a bit of work left on this, right now we only detect the common ones but we really need to add support to detect any variable.

**Comment by yaichenbaum** [comment]
@soumyamahunt With your recent changes, can you confirm that this issue is resolved with the latest beta builds?

**Comment by soumyamahunt** [comment]
> @soumyamahunt With your recent changes, can you confirm that this issue is resolved with the latest beta builds?

I tested all the environment variables that the op mentioned, and only "%programdata%" doesn't work (because it points to "C:\ProgramData" which is a hidden folder), everything else works. Also, in my testing I created my custom environment variable and used it and it worked.

**Comment by yaichenbaum** [comment]
@KaKi87 Sorry about the long wait on this feature, it's currently in beta testing and it should roll out to everyone else in a couple of weeks.

**Comment by KaKi87** [comment]
Thanks. How do I download beta ?

**Comment by yaichenbaum** [comment]
> Thanks. How do I download beta ?

@KaKi87 You can fill you the form that is linked to on the release page, or you can build from the source code.

**Comment by KaKi87** [comment]
You know I'm not into that xD
But if you can give me an APPX I'll absolutely try it out, otherwise I'll wait like everyone else ;)
